discretisedfield/tests/test_field.py

```diff
@@ -416,6 +416,113 @@ def test_unit(test_field):
         df.Field(mesh, nvdim=1, unit=1)
 
 
+@pytest.mark.parametrize(
+    "nvdim",
+    [1, 2, 3, 4],
+)
+def test_valid_single_value(valid_mesh, nvdim):
+    # Default
+    f = df.Field(
+        valid_mesh,
+        nvdim=nvdim,
+    )
+    assert f.valid.shape == (*valid_mesh.n, 1)
+    assert f.valid.dtype == bool
+    assert np.all(f.valid)
+
+    # Constant
+    f = df.Field(valid_mesh, nvdim=nvdim, valid=True)
+    assert f.valid.shape == (*valid_mesh.n, 1)
+    assert np.all(f.valid)
+
+    f = df.Field(valid_mesh, nvdim=nvdim, valid=False)
+    assert f.valid.shape == (*valid_mesh.n, 1)
+    assert f.valid.dtype == bool
+    assert np.all(~f.valid)
+
+
+@pytest.mark.parametrize("ndim", [1, 2, 3, 4])
+@pytest.mark.parametrize(
+    "nvdim",
+    [1, 2, 3, 4],
+)
+def test_valid_set_on_norm(ndim, nvdim):
+    mesh = df.Mesh(p1=(0,) * ndim, p2=(10,) * ndim, cell=(1,) * ndim)
+
+    def norm_func(point):
+        if np.all(point < 5):
+            return 5.0
+        else:
+            return 0
+
+    f = df.Field(mesh, nvdim=nvdim, value=(1,) * nvdim, norm=norm_func, valid="norm")
+    assert f.valid.shape == (*mesh.n, 1)
+    assert f.valid.dtype == bool
+    for idx in f.mesh.indices:
+        if all(f.mesh.index2point(idx) < 5):
+            assert f.valid[tuple(idx)][0]
```

Comment:
The extra `[0]` is a bit confusing at first (as usual). Maybe add a short comment.